### DISCUSSION TITLE
How much python do I need to learn to contribute to the project?

### DISCUSSION THREAD

**Discussion by DeltaEpsilon19498** [streamlink/streamlink#5780] [General]
### Description

I have never coded anything in my life. How much python do I have to learn to contribute to the project and maintain plugins?

**Comment by bastimeyer** [comment]
Pull requests will always get reviewed. Submitted code changes don't have to be perfect. If something isn't right, then there will either be guidance with code suggestions, or the code can be updated by any maintainer with write access to the PR branch.

Getting started with updating and/or writing new plugins varies in complexity. If you've never done anything like that and don't have any experience at all in terms of programming and the tools being used either, then this will surely take you a bit of time. But this is what you have to figure out on your own.
